client/store/planner.js

```diff
@@ -77,77 +83,173 @@ export const mutations = {
+
+          let start_hour = Number(start_time.slice(0,2))
+          let end_hour = Number(end_time.slice(0,2))
+          let start_minute = Number(start_time.slice(3,5))
+          let end_minute = Number(end_time.slice(3,5))
+          let recurr_string = 'FREQ=WEEKLY;BYDAY='
+
+          if (start_time.slice(-2) == 'pm' && start_hour > 12) {
+              start_hour += 12
+          }
+          if (end_time.slice(-2) == 'pm' && start_hour > 12) {
+              end_hour += 12
+          }
+
+          let duration_minute = 0
+
+          if (start_minute > end_minute) {
+            duration_minute = start_minute - end_minute
+            end_hour++
+          } else {
+            duration_minute = end_minute - start_minute
+          }
+          let duration_hour = end_hour - start_hour
+
+          if (start_hour > 14) {
+              start_hour -= 8
+              start_day++
+              plan_course.days.forEach(function(day) {
+                  switch(day) {
+                    case 'M':
+                      recurr_string += day_map.get('T')
+                      break;
+                    case 'T':
+                      recurr_string += day_map.get('W')
+                      break;
+                    case 'W':
+                      recurr_string += day_map.get('R')
+                      break;
+                    case 'R':
+                      recurr_string += day_map.get('F')
+                      break;
+                    case 'F':
+                      recurr_string += day_map.get('S')
+                      break;
+                    case 'S':
+                      recurr_string += day_map.get('U')
+                      break;
+                    case 'U':
+                      recurr_string += day_map.get('M')
+                      break;
+                  }
+                  recurr_string += ','
+              })  
+          } else {
+            start_hour += 7
+            plan_course.days.forEach(function(day) {
+                recurr_string += day_map.get(day) + ','
+            })
+          }
+          
+          recurr_string = recurr_string.slice(0, -1)
+          recurr_string += ';INTERVAL=1;UNTIL='
+          recurr_string += moment().year() + end_date.slice(0, end_date.indexOf('/')) + end_date.slice(end_date.indexOf('/') + 1)
+          recurr_string += 'T070000Z'
+
+          let course_ics_event = {
+              title: plan_course.title,
+              start: [moment().year(), start_date.slice(0, start_date.indexOf('/')), start_day, start_hour, start_minute],
+              duration: { hours: duration_hour, minutes: duration_minute },
+              location: plan_course.loc,
+              recurrenceRule: recurr_string
+          }
+          icsEvents.push(course_ics_event)
+      })
+
+      const { error, value } = ics.createEvents(icsEvents)
+
+      if (error) {
+        console.log('error')
+        console.log(error)
+        return
```

Comment:
On error we should probably tell the user an error happened. I'd disable the button after they click it.